### DISCUSSION TITLE
[IntelliSense] How can I turn on Tailwind-Intellisense sugesstions for VSCode in rust (*.rs) files?

### DISCUSSION THREAD

**Discussion by artgocode** [tailwindlabs/tailwindcss#6705] [Help]
Greeting!
I am tinkering with [Perseus](https://github.com/arctic-hen7/perseus) rust full stack framework and I want to use TailwindCSS for a frontend. Here is an example of a simple component:
```
#[perseus::template(AboutPage)]
#[component(AboutPage<G>)]
pub fn about_page() -> View<G> {
    view! {
        p(class="text-2xl font-bold") { "About." }
    }
}
```

Here `p(class="text-2xl font-bold") { "About." }` is analogue of `<p class="text-2xl font-bold">About.</p>`. I want to decorate `p` tag with some tailwind classes from Tailwind-Intellisense suggestions, but my suggestions list is empty.

What configuration should I provide to Tailwind-Intellisense extension for suggestions to appear?

**Comment by alexanderbluhm** [comment]
Hi @artgocode 
I pasted your code in _play.tailwindcss.com_ and the IntelliSense is working.
So, seems that you only have to include your file extension into `tailwindCSS.includeLanguages`.
You can look into [this](https://medium.com/@spaquet/tailwind-css-autocomplete-in-vscode-with-ruby-on-rails-63a5fc24a1a4) article on how to do this in detail.

**Comment by artgocode** [comment]
Thank you for a tip @alexanderbluhm , but I have no luck with a configuration yet.
Here is my settings.json excerpt:
```
// More settings here
"emmet.includeLanguages": {
    "rust": "html",
    "*.rs": "html"
},
"tailwindCSS.includeLanguages": {
    "rust": "html",
    "*.rs": "html"
},
"files.associations": {
    "*.rs": "rust"
},
// More settings here
```
Tasilwindcss suggestions are still not shown.
Also, I turned off all my extensions and left only tailwindcss-intellisense on. Still no luck!

![vscode-tailwind](https://user-images.githubusercontent.com/18162087/147297201-ef29da3c-6295-4897-9e44-36fd0a1048ec.JPG)

**Reply by alexanderbluhm** [reply]
@artgocode I tested it with a `test.rs` file and it is working with these settings for me.
Maybe a restart can solve it?
Do you have both latest extension and VSCode version?

**Reply by artgocode** [reply]
I have checked VSCode version and tailwind extension, and they are all updated to the latest versions.
![vscode-version](https://user-images.githubusercontent.com/18162087/147298686-b1108adf-663c-4e3b-8594-42dc5f490b0e.JPG)
![tailwind-intellisens-version](https://user-images.githubusercontent.com/18162087/147298687-f6b57ba7-acdd-49da-af83-ec8ba1ecc616.JPG)
Hmmm ...

**Reply by CalebEverett** [reply]
@artgocode were you able to get this going?

**Reply by CalebEverett** [reply]
That article helped - telling `tailwindCSS.includeLanguages` to treat rust files like html did the trick:
```
    "tailwindCSS.includeLanguages": {
        "rust": "html"
    },
    "files.associations": {
        "*.rs": "rust"
    }
 ```

**Reply by artgocode** [reply]
Hey there! I was not able to get tailwind vscode  extension to work properly for *.rs files.

I was tinkering with css in perseus framework. I was wondering, could I develop an app as fast as with laravel/php.

I got not so smooth experience yet developing rust web framework apps comparing to php frameworks.

But now I am forced to stop such an experiment because of my native country Ukraine has been invaded with Russian troops murdering civilians.

My goal for nearest future is to help my compatriots to make my motherland free from Russian invaders.

Wish you all have a peaceful sky above your head!

Слава Україні!

**Reply by Kiniber** [reply]
I tried out Seed today which uses a rust macro for classes that looks like this example: C!["my-clazz bg-red-500 mx-4"]
I got it to work by using the following in VSCode settings.xml:

```
"files.associations": {
        "*.rs": "rust"
    },
"tailwindCSS.includeLanguages": {
        "rust": "html"
    },
"tailwindCSS.experimental.classRegex": [
        "C!\\[\"([^\\]]+)\"\\]"
    ]
```
The regex means:
Letter C followed by ! followed by [ followed by "
followed by ( which is the start of the capturing group and not a character in the text
followed by multiple characters that are not [
followed by ) which is the end of the capturing group and not a character in the text
followed by " followed by ]

Maybe it can be solved easier though, but I'm glad I got it to work somehow and I hope it helps.

**Comment by almenjonatan** [comment]
These settings worked for me with yew
```
// More settings here
"emmet.includeLanguages": {
    "rust": "html",
    "*.rs": "html"
},
"tailwindCSS.includeLanguages": {
    "rust": "html",
    "*.rs": "html"
},
"files.associations": {
    "*.rs": "rust"
},
// More settings here

```

**Reply by meisme-dev** [reply]
I'm also using yew but this does not work for me

**Comment by Salman-Sali** [comment]
For anyone else, for whom its still not working, like me, please check if you have tailwind.config.js file in your main dir.
